File: java/src/main/java/info/dmerej/contacts/ContactsGenerator.java
package info.dmerej.contacts;

import java.util.Arrays;
import java.util.stream.IntStream;
import java.util.stream.Stream;

public class ContactsGenerator {
  public Stream<Contact> generateContacts(int count) {
    // TODO: generate a *lot* of contacts instead of just 3
    Contact[] contacts = new Contact[]{
      new Contact("name-1", "email-1@domain.tld"),
      new Contact("name-2", "email-2@domain.tld"),
      new Contact("name-3", "email-3@domain.tld"),
    };
    return Arrays.stream(contacts);
  }
}

File: cs/Contacts/ContactsGenerator.cs
namespace Contacts;

public class ContactsGenerator
{
  public ContactsGenerator()
  {
  }

  public List<Contact> GenerateContacts(int count)
  {
    // TODO: generate `count` contacts without
    // filling up memory - maybe using IEnemerable ?
    Contact c1 = new("name-1", "email-1@domain.tld");
    Contact c2 = new("name-2", "email-2@domain.tld");
    Contact c3 = new("name-3", "email-3@domain.tld");
    return new List<Contact> { c1, c2 };
  }
}